src/history.js

```diff
@@ -187,14 +187,14 @@ class History extends Emitter {
     if (this.size <= 0) {
       this.begin()
       return
-    } else if (!next) {
+    } else if (action === this.head) {
       next = this.head = parent
     } else if (action === this.root) {
       this.root = next
     }
 
     if (!action.disabled) {
-      this.reconcile(next)
+      this.reconcile(this.head)
```

Comment:
We need to reconcile history at the point of change. What happens in the following scenario?

```
[a] -> [b] -> [c]
```

1. Remove a

With this change, history would reconcile at _`c`_, when it needs to be _`_b`. Do I have that right?